src/core/Akka.Remote.TestKit/BarrierCoordinator.cs

```diff
@@ -564,7 +555,7 @@ protected void InitFSM()
                 {
                     case EnterBarrier barrier:
                         if (barrier.Name != currentBarrier)
-                            throw new WrongBarrierException(barrier.Name, Sender, @event.StateData);
+                            throw new WrongBarrierException(barrier.Name, Sender, barrier.Role, @event.StateData);
```

Comment:
Passing in the new `RoleName` parameter expected by the `WrongBarrierException`.